src/core/provider.h

```diff
@@ -200,6 +236,36 @@ class HTTPInferRequestProvider : public InferRequestProvider {
   std::unordered_map<std::string, size_t> input_map_;
 };
 
+//
+// Inference input provider for an internal inference request
+//
+class SystemMemoryInferRequestProvider : public InferRequestProvider {
+ public:
+  // Create a SystemMemoryInferRequestProvider object. The 'blocks' object is
+  // a mapping from input name to memory block that stores data for the input.
+  static Status Create(
+      const InferenceBackend& is, const std::string& model_name,
+      const int64_t model_version, const InferRequestHeader& request_header,
+      const std::unordered_map<std::string, std::shared_ptr<SystemMemoryBlock>>&
+          blocks,
+      std::shared_ptr<SystemMemoryInferRequestProvider>* infer_provider);
+
+  Status GetNextInputContent(
+      const std::string& name, const void** content, size_t* content_byte_size,
+      bool force_contiguous) override;
+
+ private:
+  SystemMemoryInferRequestProvider(
+      const std::string& model_name, const int64_t version)
+      : InferRequestProvider(model_name, version)
+  {
+  }
+
+  std::unordered_map<
+      std::string, std::pair<std::shared_ptr<SystemMemoryBlock>, bool>>
+      input_map_;
```

Comment:
Be consistent with naming. Here you say "input_map_", below it is output_block_. Other places you just say "blocks".